src/platform/packages/shared/kbn-lens-common/locator_types.ts

```diff
@@ -41,7 +41,11 @@ export interface LensShareableState {
   /**
    * Set the visualization configuration
    */
-  visualization: { activeId: string | null; state: unknown } & SerializableRecord;
+  visualization: {
+    activeId: string | null;
+    state: unknown;
+    selectedLayerId: string | null;
+  } & SerializableRecord;
```

Comment:
qq: why not to use directly the `VisualizationState` interface here?

```suggestion
  visualization: VisualizationState & SerializableRecord;
```